core/arch/arm/plat-hikey/main.c

```diff
@@ -93,3 +108,166 @@ void console_flush(void)
 {
 	pl011_flush(console_base());
 }
+
+static vaddr_t peripherals_base(paddr_t pa)
+{
+	void *va;
+
+	if (cpu_mmu_enabled()) {
+		va = phys_to_virt(pa, MEM_AREA_IO_NSEC);
```

Comment:
`peripherals_base()` might be used (in the future) for real usecases, not only periph inits and tests.
I would recommend to use the usual trick to prevent translation each time a periph iomem address is required:
```
	static vaddr_t va __early_bss;

	if (cpu_mmu_enabled()) {
		if (!va)
			va = phys_to_virt(pa, MEM_AREA_IO_NSEC);
		return va;
	}
	return pa;
```